src/ui/units/workbooks/store/actions/index.ts

```diff
@@ -315,6 +417,91 @@ export const renameEntry = ({
     };
 };
 
+type ChangeFavoriteEntryLoadingAction = {
+    type: typeof CHANGE_FAVORITE_ENTRY_LOADING;
+};
+type ChangeFavoriteEntrySuccessAction = {
+    type: typeof CHANGE_FAVORITE_ENTRY_SUCCESS;
+    data: AddFavoriteResponse | DeleteFavoriteResponse;
+};
+type ChangeFavoriteEntryFailedAction = {
+    type: typeof CHANGE_FAVORITE_ENTRY_FAILED;
+    error: Error;
+};
+type ChangeFavoriteEntryInlineAction = {
+    type: typeof CHANGE_FAVORITE_ENTRY_INLINE;
+    data: AddFavoriteResponse | DeleteFavoriteResponse;
+};
+type ChangeFavoriteEntryAction =
+    | ChangeFavoriteEntryLoadingAction
+    | ChangeFavoriteEntrySuccessAction
+    | ChangeFavoriteEntryFailedAction
+    | ChangeFavoriteEntryInlineAction;
+
+export const changeFavoriteEntry = ({
+    entryId,
+    isFavorite = false,
+    updateInline = false,
+}: {
+    entryId: string;
+    isFavorite?: boolean;
+    updateInline: boolean;
+}) => {
+    return (dispatch: WorkbooksDispatch) => {
+        const thenHandler = (data: AddFavoriteResponse | DeleteFavoriteResponse) => {
+            dispatch({
+                type: CHANGE_FAVORITE_ENTRY_SUCCESS,
+                data,
+            });
+
+            if (updateInline) {
+                dispatch({
```

Comment:
I suggest we leave it as it is now, since it is better to wait for the server's response and then change the value, otherwise we may get flickers when we changed inline and an error came from the server, we will have to return the value back, it is better to think through the logic with the loader